src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/ConditionalConfig.cs

```diff
@@ -47,5 +47,174 @@ public IEnumerable<IOperationProvider> Process(IComponentManager componentManage
 
             yield return new Conditional(tokenVariants, wholeLine, trim, evaluator, id);
         }
+
+        public static IReadOnlyList<IOperationProvider> ConditionalSetup(ConditionalType style, string evaluatorType, bool wholeLine, bool trimWhiteSpace, string id)
+        {
+            List<IOperationProvider> setup;
+
+            switch (style)
+            {
+                case ConditionalType.Xml:
+                    setup = XmlConditionalSetup(evaluatorType, wholeLine, trimWhiteSpace, id);
+                    break;
+                case ConditionalType.Razor:
+                    setup = RazorConditionalSetup(evaluatorType, wholeLine, trimWhiteSpace, id);
+                    break;
+                case ConditionalType.CWithComments:
+                    setup = CStyleWithCommentsConditionalSetup(evaluatorType, wholeLine, trimWhiteSpace, id);
+                    break;
+                case ConditionalType.CNoComments:
+                    setup = CStyleNoCommentsConditionalSetup(evaluatorType, wholeLine, trimWhiteSpace, id);
+                    break;
+                case ConditionalType.CBlockComments:
+                    setup = CBlockCommentConditionalSetup(evaluatorType, wholeLine, trimWhiteSpace, id);
+                    break;
+                default:
+                    throw new Exception($"Unrecognized conditional type {style}");
+            }
+
+            return setup;
+        }
+
+        public static List<IOperationProvider> XmlConditionalSetup(string evaluatorType, bool wholeLine, bool trimWhiteSpace, string id)
+        {
+            // This is the operationId (flag) for the balanced nesting.
+            string commentFixingOperationId = "Fix pseudo comments";
```

Comment:
These Ids should probably be generated on the off chance that we'd ever end up having multiple block comment type conditionals in the same file